x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/optimizer/Optimizer.java

```diff
@@ -1130,7 +1130,8 @@ protected LogicalPlan rule(LogicalPlan plan) {
 
             plan.forEachDown(a -> {
```

Comment:
@matriv I don't think this whole thing is still needed from 1131 till line 1141.